File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/JsonFieldArrayWrapper.java
package org.codinjutsu.tools.nosql.commons.view.add;

import com.google.gson.Gson;
import com.google.gson.JsonArray;
import org.codinjutsu.tools.nosql.commons.model.internal.layer.DatabaseElement;

import static org.codinjutsu.tools.nosql.commons.model.internal.json.JsonHelperKt.convert;

public class JsonFieldArrayWrapper extends JTextFieldWrapper {

    @Override
    public DatabaseElement getValue() {
        return convert(new Gson().fromJson(component.getText(), JsonArray.class));
    }
}

File: src/main/java/org/codinjutsu/tools/nosql/commons/view/add/JsonFieldArrayWrapper.kt
package org.codinjutsu.tools.nosql.commons.view.add

import com.google.gson.Gson
import com.google.gson.JsonArray
import com.intellij.openapi.project.Project
import org.codinjutsu.tools.nosql.commons.model.internal.toDatabaseElement

internal class JsonFieldArrayWrapper(project: Project) : JsonEditorWrapper(project) {

    override fun getValue() =
            Gson().fromJson(editor.document.text, JsonArray::class.java).toDatabaseElement()

    override fun reset() {
        editor.document.setText("[]")
    }
}
